port/zos390/omrvmem.c

```diff
@@ -1352,9 +1366,18 @@ omrvmem_find_valid_page_size(struct OMRPortLibrary *portLibrary, uintptr_t mode,
 		uintptr_t defaultLargePageSize = 0;
 		uintptr_t defaultLargePageFlags = OMRPORT_VMEM_PAGE_FLAG_NOT_USED;
 
-		/* If the page type is NOT_USED (as with the -Xlp<size> options)
-		 * the legacy behavior is to default to FIXED pages.
-		 */
+		/* If the page type is PREFER_PAGEABLE try to get 1M pageable large pages. Otherwise, use a fixed large page. */
+		if ((OMRPORT_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE == validPageFlags) && (ONE_M == validPageSize)) {
```

Comment:
I am not a fan of hiding operations inside Macro but (because code uses this style already) you can define new Macro `IS_VMEM_PAGE_FLAG_PAGEABLE_PREFERABLE()`  (similar to `IS_VMEM_PAGE_FLAG_NOT_USED()`)

Reply:
Done. See 923728f8c7f40a80bc81474ea4fca1b85c26ea21